src/ln/channelmanager.rs

```diff
@@ -1646,6 +1644,22 @@ impl ChannelMessageHandler for ChannelManager {
 			};
 
 		let mut channel_state_lock = self.channel_state.lock().unwrap();
+		{
+			if next_hop_data.hmac != [0;32] {
+				// check doesn't apply if we are last node (check happens here due to lock optimizations)
+				let chan = channel_state_lock.by_id.get(&msg.channel_id).unwrap();
```

Comment:
This is still wrong - we should be doing the fee checks against the place we're sending the payment (ie pending_forward_info.short_channel_id ie the channel that is looked up in the next if statement.